src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -665,6 +672,407 @@ void M1()
+            (var operation, _) = GetOperationAndSyntaxForTest<VariableDeclarationSyntax>(compilation);
+            var declaration = (IVariableDeclarationOperation)operation;
+            Assert.Equal(2, declaration.Children.Count());
+            Assert.Equal(OperationKind.Literal, declaration.Children.First().Kind);
+            Assert.Equal(OperationKind.VariableDeclarator, declaration.Children.ElementAt(1).Kind);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_VerifyInvalidDimensions()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        int[/*<bind>*/10/*</bind>*/] x;
+    }
+}
+";
+
+            var compilation = CreateEmptyCompilation(source);
+            (var operation, _) = GetOperationAndSyntaxForTest<LiteralExpressionSyntax>(compilation);
+            var declaration = (ILiteralOperation)operation;
+            Assert.True(declaration.ConstantValue.HasValue);
+            Assert.Equal(10, declaration.ConstantValue.Value);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_TestSemanticModel()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        int[10] x;
+        int[M2()]
+    }
+
+    int M2() => 42;
+}
+";
+
+            var tree = Parse(source, options: TestOptions.Regular);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+            var literalExpr = nodes.OfType<LiteralExpressionSyntax>().ElementAt(0);
+
+            Assert.Equal(@"10", literalExpr.ToString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(literalExpr).Type.ToTestDisplayString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(literalExpr).ConvertedType.ToTestDisplayString());
+            Assert.Equal(Conversion.Identity, model.GetConversion(literalExpr));
+
+            var invocExpr = nodes.OfType<InvocationExpressionSyntax>().ElementAt(0);
+
+            Assert.Equal(@"M2()", invocExpr.ToString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(invocExpr).Type.ToTestDisplayString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(invocExpr).ConvertedType.ToTestDisplayString());
+            Assert.Equal(Conversion.Identity, model.GetConversion(invocExpr));
+
+            var invocInfo = model.GetSymbolInfo(invocExpr);
+            Assert.NotNull(invocInfo.Symbol);
+            Assert.Equal(SymbolKind.Method, invocInfo.Symbol.Kind);
+            Assert.Equal("M2", invocInfo.Symbol.MetadataName);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_OutVarDeclaration()
```

Comment:
>IVariableDeclaration_InvalidIgnoredDimensions_OutVarDeclaration [](start = 20, length = 63)

For the ```out var``` and pattern declarations, please add tests to src\Compilers\CSharp\Test\Semantic\Semantics\OutVarTests.cs and src\Compilers\CSharp\Test\Semantic\Semantics\PatternMatchingTests_Scope.cs. The tests should verify that the locals are added to the right scope and proper shadowing is done for duplicate declarations, SemanticModel behaves as expected. There are many test examples in the files.